artifactory.py

```diff
@@ -510,8 +509,8 @@ def splitroot(self, part, sep=sep):
                 path = self._get_path(part)
                 drv = part.rpartition(path)[0]
                 path_parts = path.strip(sep).split(sep)
-                root = sep + path_parts[0] + sep
-                rest = sep.join(path_parts[1:])
+                root = sep
+                rest = sep.join(path_parts[0:])
```

Comment:
Are all these changes needed for Python 3.12 compatibility?  

Can we limit the PR to just the necessary updates? This will make it easier to debug if it causes any regression issues.

Reply:
I tried addressing this in the PR description because I completely understand this might be a blocker. Let me try explaining it again, and I'll be happy to try any suggestion you may have to avoid this set of changes (I couldn't think of one, apologies).

When I tried just doing the changes for 3.12 compatibility, one of the tests that broke was the one that relied on this being true:

```python
ArtifactoryPath('http://b.com/artifactory/repo/path/file.txt').is_relative_to('http://b.com/artifactory')
```

This statement returns true in python3.11 but returns false in python3.12, and my guess is that we are relying on undefined behavior, based on how PureWindowsPath works.

```python
# python 3.12
>>> PureWindowsPath("C://abc/def.txt").is_relative_to("C:")
False
```

So I felt I had two choices: Override most `PurePath` methods for artifactory, so it would behave as it used to in python3.11, or change ArtifactoryPath to consider `/` as root, instead of the repo. I thought the first option would be a maintenance nightmare, so that is why I went with the second one, but I could be wrong.

Happy to hear your thoughts on the subject!

(Also, all the changes for this new behavior are part of the first commit, happy to push that as a separate pull request if you think it would make things easier)